examples/magento-graphcms/components/GraphCMS/RowColumnOne/RowColumnOne.tsx

```diff
@@ -1,17 +1,35 @@
-import { RichText, RichTextProps } from '@graphcommerce/graphcms-ui'
-import { ColumnOne } from '@graphcommerce/next-ui'
+import { RichTextProps } from '@graphcommerce/graphcms-ui'
 import type { RowColumnOneFragment } from './RowColumnOne.gql'
+import { Default, Message } from './variant'
+
+type VariantRenderer = Record<
+  NonNullable<RowColumnOneFragment['rowColumnOneVariant']>,
+  React.VFC<RowColumnOneFragment>
+>
 
 export type RowColumnOneProps = RowColumnOneFragment & {
+  renderer?: Partial<VariantRenderer>
   richTextOne?: Omit<RichTextProps, 'raw'>
 }
 
+const defaultRenderer: Partial<VariantRenderer> = {
+  Default,
+  Message,
+}
+
 export function RowColumnOne(props: RowColumnOneProps) {
-  const { colOne, richTextOne } = props
+  const { renderer, rowColumnOneVariant, ...RowColumnOneProps } = props
+  const mergedRenderer = { ...defaultRenderer, ...renderer } as VariantRenderer
+
+  if (!rowColumnOneVariant) return <Default {...RowColumnOneProps} />
+
+  const RenderType =
+    mergedRenderer?.[rowColumnOneVariant] ??
+    (() => {
+      if (process.env.NODE_ENV !== 'production')
+        return <>renderer for {rowColumnOneVariant} not found</>
+      return null
+    })
```

Comment:
Something like this?

```suggestion
  const { renderer, rowColumnOneVariant = 'Default', ...RowColumnOneProps } = props
  const mergedRenderer = { ...defaultRenderer, ...renderer } as VariantRenderer

  const RenderType = mergedRenderer?.[rowColumnOneVariant]
  if (!RenderType) return process.env.NODE_ENV !== 'production' ? <>renderer for {rowColumnOneVariant} not found</> : null
```

Reply:
`rowColumnOneVariant` can be `null`, so setting a default value like this won't work. However, I do like setting `rowColumnOneVariant` to `'Default'` more than an early return. I've updated the PR to use that instead.